pkg/apis/apis/v1alpha1/api_cel_test.go

```diff
@@ -0,0 +1,216 @@
+		{
+			name: "valid non core resource",
+			claim: map[string]interface{}{
+				"group":        "new.core.resources",
+				"resource":     "fakeresources",
+				"identityHash": "fakehashhere",
+			},
+			validBinding: true,
+		},
+	}
+
+	validator, s := getValidatorForPermissionClaim(t, "../../../../config/crds/apis.kcp.dev_apibindings.yaml", "acceptedPermissionClaims")
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validator.Validate(field.NewPath("root"), &s, tc.claim)
+			if len(errs) == 0 && !tc.validBinding {
+				t.Error("No errors were found, but should be invalid binding")
+				return
+			}
+			if len(errs) > 0 && tc.validBinding {
+				t.Errorf("found errors: %v but should be valid binding", errs.ToAggregate().Error())
+				return
+			}
+		})
+	}
+}
+
+// TestAPIBindingPermissionClaimCELValidation will validate the permission claims for an otherwise valid APIBinding.
+func TestAPIExportPermissionClaimCELValidation(t *testing.T) {
+	testCases := []struct {
+		name         string
+		claim        map[string]interface{}
+		validBinding bool
+	}{
+		{
+			name: "valid",
+			claim: map[string]interface{}{
+				"group":    "",
+				"resource": "configmaps",
+			},
+			validBinding: true,
+		},
+		{
+			name: "invalid core resource",
+			claim: map[string]interface{}{
+				"group":    "",
+				"resource": "fakeresources",
+			},
+			validBinding: false,
+		},
+		{
+			name: "invalid non core resource",
+			claim: map[string]interface{}{
+				"group":    "new.core.resources",
+				"resource": "fakeresources",
+			},
+			validBinding: false,
+		},
+		{
+			name: "valid non core resource",
+			claim: map[string]interface{}{
+				"group":        "new.core.resources",
+				"resource":     "fakeresources",
+				"identityHash": "fakehashhere",
+			},
+			validBinding: true,
+		},
+	}
+
+	validator, s := getValidatorForPermissionClaim(t, "../../../../config/crds/apis.kcp.dev_apiexports.yaml", "permissionClaims")
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validator.Validate(field.NewPath("root"), &s, tc.claim)
+			if len(errs) == 0 && !tc.validBinding {
+				t.Error("No errors were found, but should be invalid binding")
+				return
+			}
+			if len(errs) > 0 && tc.validBinding {
+				t.Errorf("found errors: %v but should be valid binding", errs.ToAggregate().Error())
+				return
+			}
+		})
+	}
+}
+
+func withRule(s schema.Structural, rule string) schema.Structural {
+	s.Extensions.XValidations = apiextensionsv1.ValidationRules{
+		{
+			Rule: rule,
+		},
+	}
+	return s
+}
+
+func primitiveType(typ, format string) schema.Structural {
+	result := schema.Structural{
+		Generic: schema.Generic{
+			Type: typ,
+		},
+	}
+	if len(format) != 0 {
+		result.ValueValidation = &schema.ValueValidation{
+			Format: format,
+		}
+	}
+	return result
+}
+
+func getValidatorForPermissionClaim(t *testing.T, crdFilePath string, specClaimName string) (*cel.Validator, schema.Structural) {
```

Comment:
we will get more uses of this func. Why not make it generic enough?

I.e. make the json path inside the schema a parameter.

Reply:
One of the big reasons is that I don't think that it needs to do it, and why spend the extra time because I do have other things that I have to get done. 

Are there plans for other CELs? We would have to have some different thing to get the schema, and it would probably need to be broken up. 

I wonder if this is a follow-up?